crates/turbopack-core/src/source_map/mod.rs

```diff
@@ -245,10 +306,78 @@ impl SourceMap {
         };
         Ok(OptionToken(token).cell())
     }
+
+    #[turbo_tasks::function]
+    pub async fn trace(self: Vc<Self>, other: Vc<SourceMap>) -> Result<Vc<SourceMap>> {
+        let own_map = match &*self.await? {
+            Self::Regular(m) => m.clone(),
+            Self::Sectioned(m) => m.flatten().await?,
+        };
+
+        let mut builder = sourcemap::SourceMapBuilder::new(own_map.get_file());
+        let tokens = other
+            .tokens()
+            .await?
+            .iter()
+            .map(|other_token| async move {
+                let other_token = match other_token {
+                    Token::Synthetic(_) => panic!("Unexpected synthetic token"),
```

Comment:
It's not unusual for a map to contain synthetic tokens, and we should preserve them.